### PR TITLE
experimental search input: Better context: filter handling on paste

### PR DESCRIPTION
**Pull request by fkling**
In the common case of pasting a new query with context: filter into the input containing only a context: filter, the existing filter is removed.

## Test plan

Manual testing, unit test.

## App preview:

- [Web](https://sg-web-fkling-search-input-context-paste.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Review by rrhyne — APPROVED** [review]
Thanks Felix!

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.50 kb) | 0.00% (+0.50 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/fkling/search-input-context-paste/compare-report.html;tab=live_object) for a full comparison between the commits 3d01edb and 43267cc or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/branded/src/search-ui/input/experimental/codemirror/searchcontext.test.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { EditorState } from '@codemirror/state'
+
+import { SearchPatternType } from '@sourcegraph/shared/src/graphql-operations'
+
+import { parseInputAsQuery } from '../../codemirror/parsedQuery'
+
+import { overrideContextOnPaste } from './searchcontext'
+
+describe('overrideContextOnPaste', () => {
```
Comment (philipp-spiess) [inline client/branded/src/search-ui/input/experimental/codemirror/searchcontext.test.ts:9]: me like test cases

**Review by philipp-spiess — APPROVED** [review]

File: client/branded/src/search-ui/input/experimental/codemirror/searchcontext.test.ts (hunk 1 above)
Comment (fkling) [reply client/branded/src/search-ui/input/experimental/codemirror/searchcontext.test.ts:9]: Me too! 🙂

**Pull request merged**